src/fec_base.h

```diff
@@ -576,52 +582,40 @@ bool FecCode<T>::decode_bufs(
     }
 
     vec::Vector<T> words(gf, n_words);
-    vec::Vector<T> fragments_ids(gf, n_words);
     vec::Vector<T> output(gf, n_data);
 
+    DecodeContext<T>* context = init_context_dec(&fragments_ids);
```

Comment:
Should be a `unique_ptr`, then you don't need to delete l.637.

In fact, your code will leak memory if an exception is raised anywhere between here and your delete so you should really use an unique_ptr here, to have a memory handling that is exception-safe

Reply:
yes, it's done